src/dstack/_internal/server/services/jobs/__init__.py

```diff
@@ -316,3 +327,45 @@ def group_jobs_by_replica_latest(jobs: List[JobModel]) -> Iterable[Tuple[int, Li
             *_, latest_job_submission = job_submissions
             replica_jobs.append(latest_job_submission)
         yield replica_num, replica_jobs
+
+
+async def detach_volumes_from_instance(
+    project: ProjectModel,
+    instance: InstanceModel,
+    jpd: JobProvisioningData,
+):
+    # TODO: Volumes lock
```

Comment:
Is it ok to merge without it?

Reply:
No need for a lock there actually since volumes in use cannot be deleted.